src/linear_algebra/orientation.lean

```diff
@@ -0,0 +1,413 @@
+    have hzzz := ne_of_gt (add_pos_of_pos_of_nonneg hr₁ (mul_nonneg hr₂.le hr)),
+    simpa [ne_of_gt (add_pos_of_pos_of_nonneg hr₁ (mul_nonneg hr₂.le hr)),
+           -mul_neg_eq_neg_mul_symm] using h },
+  { exact λ h, same_ray_pos_smul_right v h }
+end
+
+/-- A multiple of a nonzero vector is in the same ray as that vector if and only if that multiple
+is positive. -/
+@[simp] lemma same_ray_smul_left_iff [no_zero_smul_divisors R M] {v : M} (hv : v ≠ 0) (r : R) :
+  same_ray R (r • v) v ↔ 0 < r :=
+begin
+  rw (symmetric_same_ray R M).iff,
+  exact same_ray_smul_right_iff hv r
+end
+
+/-- The negation of a nonzero vector is in the same ray as a multiple of that vector if and
+only if that multiple is negative. -/
+@[simp] lemma same_ray_neg_smul_right_iff [no_zero_smul_divisors R M] {v : M} (hv : v ≠ 0)
+  (r : R) : same_ray R (-v) (r • v) ↔ r < 0 :=
+begin
+  rw [←same_ray_neg_iff, neg_neg, ←neg_smul, same_ray_smul_right_iff hv (-r)],
+  exact right.neg_pos_iff
+end
+
+/-- A multiple of a nonzero vector is in the same ray as the negation of that vector if and
+only if that multiple is negative. -/
+@[simp] lemma same_ray_neg_smul_left_iff [no_zero_smul_divisors R M] {v : M} (hv : v ≠ 0)
+  (r : R) : same_ray R (r • v) (-v) ↔ r < 0 :=
+begin
+  rw [←same_ray_neg_iff, neg_neg, ←neg_smul, same_ray_smul_left_iff hv (-r)],
+  exact left.neg_pos_iff
+end
+
+namespace basis
+
+variables [fintype ι]
+
+/-- The orientations given by two bases are equal if and only if the determinant of one basis
+with respect to the other is positive. -/
+lemma orientation_eq_iff_det_pos (e₁ e₂ : basis ι R M) :
+  e₁.orientation = e₂.orientation ↔ 0 < e₁.det e₂ :=
+by rw [basis.orientation, basis.orientation, ray_eq_iff,
+       e₁.det.eq_smul_basis_det e₂, alternating_map.smul_apply, basis.det_self, smul_eq_mul,
+       mul_one, same_ray_smul_left_iff e₂.det_ne_zero (_ : R)]
+
+/-- Given a basis, any orientation equals the orientation given by that basis or its negation. -/
+lemma orientation_eq_or_eq_neg (e : basis ι R M) (x : orientation R M ι) :
+  x = e.orientation ∨ x = -e.orientation :=
+begin
+  rw [basis.orientation, ←x.some_vector_ray, ray_eq_iff, ←ray_neg, ray_eq_iff,
+      x.some_vector.eq_smul_basis_det e],
+  rcases lt_trichotomy (x.some_vector e) 0 with h|h|h,
+  { right,
+    exact (same_ray_neg_smul_left_iff e.det_ne_zero (_ : R)).2 h },
+  { have hz := x.some_vector.eq_smul_basis_det e,
+    rw h at hz,
+    simpa using hz },
+  { left,
+    exact (same_ray_smul_left_iff e.det_ne_zero (_ : R)).2 h }
+end
+
+end basis
+
+namespace module.ray
+
+/-- A ray does not equal its own negation. -/
+lemma ne_neg_self [no_zero_smul_divisors R M] (x : module.ray R M) : x ≠ -x :=
+begin
+  intro h,
+  induction x using module.ray.ind,
+  rw [←ray_neg, ray_eq_iff] at h,
+  rcases h with ⟨r₁, r₂, hr₁, hr₂, h⟩,
+  rw [smul_neg, ←neg_smul, ←sub_eq_zero, ←sub_smul] at h,
+  simpa [ne_of_gt (add_pos hr₁ hr₂)] using h
```

Comment:
These might be worth extracting to a lemma:
```lean
lemma eq_zero_of_same_ray_self_neg [no_zero_smul_divisors R M] {v₁ : M} (h : same_ray R v₁ (-v₁)) :
  v₁ = 0 :=
begin
  rcases h with ⟨r₁, r₂, hr₁, hr₂, h⟩,
  rw [smul_neg, ←neg_smul, ←sub_eq_zero, ←sub_smul, sub_neg_eq_add, smul_eq_zero] at h,
  exact h.resolve_left (add_pos hr₁ hr₂).ne',
end
```

Reply:
Done, and also moved both those lemmas from `linear_ordered_comm_ring` to `ordered_comm_ring` since they didn't in fact need a linear order.